### ISSUE TITLE
Add index (i) to the handler call

### ISSUE THREAD

**Issue by agrohs** [sergelerator/async-batch#6]
Wondering if it would be possible to add the index to the handler call along w the task (so it would behave just a drop closer to a normal es6 loop)?

https://github.com/sergelerator/async-batch/blob/10614d56e5bf5ee3b70ef434672f504e2f0e6b46/src/index.ts#L22

Something similar to one of the following would be ideal such that each handler function would get called w/ the task as well as the counter (in case it needed it for anything)
```
    while (i < tasks.length) {
      const taskId = i;
      i++;
      results[taskId] = await handler(tasks[taskId], i);
    }
```
OR
```
    while (i < tasks.length) {
      const taskId = i;
      results[taskId] = await handler(tasks[taskId], i);
      i++;
    }
```

**Comment by agrohs** [comment]
@sergelerator  - Updated full version below including that update (as well as some performance tweaks)!
```
const asyncBatch = async <TaskType, ResultType>(
  tasks: TaskType[],
  handler: (task: TaskType, index: number, workerIndex: number) => Promise<ResultType>,
  _workers: number,
): Promise<ResultType[]> => {
  const length = Math.max(Math.floor(Math.min(_workers, tasks.length)), 1)

  const results: ResultType[] = [];
  let index = 0;
  await Promise.all(Array.from({ length })
    .map(async (_worker, workerIndex) => {
      do {
        const taskIndex = index;
        index++;
        results[taskIndex] = await handler(tasks[taskIndex], taskIndex, workerIndex);
      } while (index < tasks.length);
    })
  );
  return results;
};

export default asyncBatch;
```

**Comment by sergelerator** [comment]
Hey @agrohs ! Sorry it took me so long to reply, I just found the notification about this issue. I'll find the time to include the change to send the index to the handler call today and publish a new version, I'm also open to receiving PRs, thank you!

**Comment by agrohs** [comment]
Sounds good, no worries @sergelerator! I'll submit a PR with it as soon as I have a few.

**Comment by sergelerator** [comment]
This is now published in version 1.1.1
